pyiron_atomistics/atomistics/structure/analyse.py

```diff
@@ -56,6 +313,93 @@ def __init__(self, structure):
             structure (:class:`pyiron.atomistics.structure.atoms.Atoms`): reference Atom structure.
         """
         self._structure = structure
+        self._interstitials = None
+
+    @property
+    def interstitials(self):
+        if self._interstitials is None:
+            raise AssertionError('Initialize interstitials via get_interstitials()')
+        return self._interstitials
```

Comment:
In line with my cosmetic comment above
```suggestion
        if self._interstitials is None:
            self._interstitials = Interstitials(…)
        return self._interstitials
```

I wouldn't really mind dropping this and forcing people to make an instance first and then use, like we do with the neighborhood.

Reply:
Here in this case we have to give a number for `num_neighbors`. I'm not really sure if we should assume a number...

Reply:
Ok, I had a couple false starts to commenting on this issue, but I think I understand now what's happening. I agree with Marvin.

Things I don't like right now:
- The discrepancy between `Atoms.get_neighbors` and `Analyze.get_interstitials`
- `Analyze.get_interstitials` returns two different class types (depending on a boolean flag) 
  - and by default it's *not* the type `Interstitials`(?!?!?!)
  - And then after it's been called `Analyze.get_interstitials` and `Analyze.interstitials` return different things
- I feel like this compounds on top of the already confusing interplay of `Interstitials.positions` vs `Interstitials.get_positions`

Reply:
I don't really understand what you guys mean with the discrepancy between `Atoms.get_neighbors` and `Analyse.get_interstitials`, because `get_neighbors` gives the information on the neighbourhood for each atom (i.e. `n_atoms x n_neighbors`), while `get_interstitials` is about giving interstitial positions. In total, I would rather expect `Interstitials` to be like `Atoms` but not like `Neighbors`.

Reply:
Oh you mean like `get_neighbors` returns `Neighbors` and `get_interstitials` doesn't? If that's the case I agree that it's maybe something that has to be changed.

Reply:
> Oh you mean like get_neighbors returns Neighbors and get_interstitials doesn't return Interstitials? If that's the case I agree that it's maybe something that has to be changed.

Sorry, missed this second comment during my initial reply. Yep! Cool, I think we're on a sufficiently similar page then.